src/clients/twitch.ts

```diff
@@ -12,128 +15,117 @@ function registerTwitch(): void
-                }, genericPromiseError);
-            }
+            if(!bridge.targetDiscordChannel)
+                throw new Error('Cannot find Discord channel.');
 
-            //If we reach this and we're looking for this very message, then we have a chance to re-bind the message that we tried to send via discord
-            else
+            // We should (hopefully) not get stuck in a loop here due to our
+            // checks in discord.ts
+            bridge.targetDiscordChannel.send(`[t][${ user }] ${ message }`).then((discordMessage: Message<boolean>) =>
             {
+                //Discord actually stores message object after the promise is fullfilled (unlike twitch), so we can just create this object on the fly
 
-                //Record that last given userstate ID (specific to bots)
-                //Set the message to contain the special value: botUserStateId
-                if(userState.id)
-                    bridge.lastUserStateMsg.userState.botUserStateId = userState.id;
+                //Map both of these results for later querying. Eventually these will go away as we're deleting messages we don't care about anymore.
+                const twitchMessage = new twitchMsg(message, false, userState, channel);
+                const listNode = bridge.messageLinkdListInterface.addNode(new conjoinedMsg(discordMessage, [twitchMessage]));
 
-                else
-                {
-                    console.log('Got the userstate message with no ID -_-');
+                bridge.discordTwitchCacheMap.set(twitchMessage, listNode);
+                bridge.discordTwitchCacheMap.set(discordMessage, listNode);
 
-                    //[trailing bot message] This is a special method of recording the last twitch message because we need to have a method of removing this message after the fact
-                    const twitchMessage = new twitchMsg(msg, self, userState, channel);
-                    bridge.lastUserStateMsg = twitchMessage;
+                //Count upwards and delete the oldest message if need be
+                manageMsgCache();
+            }, genericPromiseError);
 
-                    //Adding an node intentionally without a discord message because this is the buffer message.
-                    const listNode = bridge.messageLinkdListInterface.addNode(new conjoinedMsg(undefined, [twitchMessage]));
-                    bridge.discordTwitchCacheMap.set(twitchMessage, listNode);
-                }
 
-                //before we override the last message, lets make sure we delete this twitch message if required (...this is jank)
-                //We wait until after the userstate message gets a bot message ID
-                if(bridge.lastUserStateMsg?.userState.cueForDelete)
-                {
-                    twitchDelete(bridge.lastUserStateMsg);
-                    manageMsgCache(bridge.discordTwitchCacheMap.get(bridge.lastUserStateMsg));
-                }
+            if(!userState.id)
```

Comment:
I haven't loaded the code yet; but I don't know if we even need this anymore

If what I heard was true in that this library fixes our trailing ID problem, we could probably get rid of this statement alltogether

Reply:
You right, I'm pretty sure we don't; Kept it due to not wanting to somehow break something, cause like the entire onMessage event handler is horrible to read ._.